src/dft/backends/mklgpu/commit.cpp

```diff
@@ -75,13 +87,37 @@ class mklgpu_commit final : public dft::detail::commit_impl<prec, dom> {
             oneapi::mkl::dft::detail::external_workspace_helper<prec, dom>(
                 config_values.workspace_placement ==
                 oneapi::mkl::dft::detail::config_value::WORKSPACE_EXTERNAL);
-        set_value(handle, config_values);
+        // Generate forward DFT descriptor.
+        set_value(*handle.first, config_values, true);
+        std::optional<mkl::exception> fwd_exception;
         try {
-            handle.commit(this->get_queue());
+            handle.first->commit(this->get_queue());
         }
         catch (const std::exception& mkl_exception) {
             // Catching the real Intel oneMKL exception causes headaches with naming.
-            throw mkl::exception("dft/backends/mklgpu", "commit", mkl_exception.what());
+            fwd_exception = mkl::exception("dft/backends/mklgpu", "commit", mkl_exception.what());
+        }
+
+        // Generate backward DFT descriptor only if required.
+        if (config_values.input_strides == config_values.output_strides) {
```

Comment:
I haven't worked on the DFT domain in a long time. How come we can be sure that we won't need the backward descriptor if the input and output strides are equal?

Reply:
We need two descriptors because we have to swap the in/output strides between the forward and backward descriptors. If the strides are the same between the two, then swapping them doesn't achieve anything, so we can use a single descriptor.

Reply:
Of course